src/dataframe/dataframe.jl

```diff
@@ -1127,11 +1127,20 @@ function categorical!(df::DataFrame,
 end
 
 """
-    append!(df1::DataFrame, df2::AbstractDataFrame)
+    append!(df1::DataFrame, df2::AbstractDataFrame; cols::Symbol=:setequal)
+    append!(df::DataFrame, x; cols::Symbol=:setequal)
 
-Add the rows of `df2` to the end of `df1`.
+Add the rows of `df2` to the end of `df1`. If the second argument `x` is
+not an `AbstractDataFrame` then it is converted using `DataFrame(x, copycols=false)`
+before being appended.
 
-Column names must be equal (including order), with the following exceptions:
+Column names of  `df1` and `df2` must be equal.
+If `cols` is `:setequal` (the default) then column names may have different orders
+and `append!` is performed by matching column names.
+If `cols` is `:orderequal` then the order of columns in `df1` and `df2` must be
```

Comment:
We should probably apply the same check for `AbstractDict` as for `push!`, right?

Reply:
added (for `Dict` as in `push!`).

I have a problem with `AbstractDicts` and `NamedTuples` that they have a dual meaning for DataFrames.jl (depending on the context they can be treated as a row or as a table), but I guess it is not crucial so we can live with it (i.e. you can both `push!` them or `append!`/convert them to a `DataFrame`).

Reply:
Yeah, that could confuse people that used `push!` when they meant `append!` or the reverse, but well... That problem also affects vectors, e.g. with `push!(Any[1], [1])`.